Reflect-cpp traits

aligirayhanozbay opened the pull request:
This PR adds traits for the [reflect-cpp](https://github.com/getml/reflect-cpp) json library

Thalhammer commented:
Hi and thanks for extending jwt-cpp.
Would it be possible to add some CI testing to ensure we don't break any of the code in the future ?
In addition the CMake scripts need modification to ensure the new files are correctly installed along the other headers.

aligirayhanozbay commented:
@Thalhammer Added tests, using FetchContent if reflectcpp is not found

aligirayhanozbay commented:
@Thalhammer I've added everything needed for CI, including

- Installation of reflect-cpp
- Building the example executable for the reflect-cpp traits
- Building the tests for the reflect-cpp traits
- Running the tests for the reflect-cpp traits

[On my fork](https://github.com/aligirayhanozbay/jwt-cpp/pull/2), the CI runs successfully for everything assosciated with the new reflect-cpp trait. Unfortunately, one CI item seems to fail ([JWT CI / coverage (pull_request)](https://github.com/aligirayhanozbay/jwt-cpp/actions/runs/17192869447/job/48770860778?pr=2)) but I see that is broken on main in the repo and that you're in the process of fixing it.

Please let me know if you'd like any further modifications.

prince-chrismc commented:
I noticed the lint ci  is missing the new json library, i am okay with it missing.

You might want to update the readme to share your excellent work!

aligirayhanozbay commented:
@Thalhammer @prince-chrismc 

Per your comments, I have

* Removed the use of Ninja
* Removed the `set(CMAKE_EXPORT_COMPILE_COMMANDS ...)` line
* Modified [.github/workflows/cmake.yml](https://github.com/Thalhammer/jwt-cpp/pull/392/files/0803017991466462afb4c9450b2eb5e9c4ec3c61#diff-cdd48abbd3eb8d1c54077449fc74a8de1f29805d2be5d8e5232b7aab76ea7a6f) to re-use the action from `.github/actions/install/reflect-cpp/action.yml`

Could you clarify what you mean by updating the readme? I don't see anything about any of the other traits or contributors in the `README.md` file.

aligirayhanozbay commented on CMakeLists.txt:
```diff
@@ -31,11 +31,13 @@ cmake_dependent_option(JWT_EXTERNAL_NLOHMANN_JSON
   "Use find_package() to locate nlohman-json required for tests and examples" OFF
   "JWT_BUILD_EXAMPLES OR JWT_BUILD_TESTS" OFF)
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL "Export compile commands" FORCE)
```
Removed

Thalhammer commented:
> Do you mean adding badges to docs/traits.md?

Yes that section used to be in the readme itself before it got too large 😅

aligirayhanozbay commented:
@Thalhammer @prince-chrismc 

Added reflect-cpp ao docs/traits.md. Any outstanding items before we can merge this?
